backend/lcfs/web/api/compliance_report/update_service.py

```diff
@@ -71,11 +92,21 @@ async def handle_status_change(
     async def handle_draft_status(self, report: ComplianceReport):
         """Handle actions when a report is set to Draft status."""
         # Implement logic for Draft status
-        pass
+        user = self.request.user
+        has_director_role = user_has_roles(
+            user, ["SUPPLIER"]
+        )
+        if not has_director_role:
+            raise HTTPException(status_code=403, detail="Forbidden.")
 
     async def handle_submitted_status(self, report: ComplianceReport):
         """Handle actions when a report is submitted."""
-
+        user = self.request.user
+        has_director_role = user_has_roles(
```

Comment:
On a fresh database this check was failing and giving a 403. Did something change in the role logic where we're testing stings against enums?

Reply:
updated